### DISCUSSION TITLE
Why uriparser support was added?

### DISCUSSION THREAD

**Discussion by ondrejholy** [FreeRDP/FreeRDP#9972] [General]
Can you please give me some insight into why the `uriparser` support was added? The relevant commit / pull request doesn't give any explanation (https://github.com/FreeRDP/FreeRDP/pull/9453/commits/34afa9be43f38dc3f2023d0849bb23fd40e0a8ce). The internal implementation is used when the dependency is disabled. So what are the benefits of using the `uriparser` instead of the internal implementation?

**Comment by akallabeth** [comment]
@ondrejholy basically to clean up the code and depend on something others have in use too.
we´ve had a lot of custom parser code integrated, but it is quite a lot of work to maintain.

**Reply by ondrejholy** [reply]
Thanks for your answer. I accept these arguments. But didn't you forget to remove the internal implementation then: https://github.com/FreeRDP/FreeRDP/blob/master/winpr/libwinpr/crt/string.c#L76-L157?

**Reply by akallabeth** [reply]
@ondrejholy it is currently there as a fallback if no `uriparser` is available.
(it is scheduled to be removed for stated reasons though)

**Reply by ondrejholy** [reply]
Can we expect that the internal implementation will be available for the whole FreeRDP 3 lifetime? I am asking because we are trying to reduce the number of new dependencies for RHEL...

**Reply by akallabeth** [reply]
yes.
we´ll be starting to drop things again if we create a FreeRDP version 4

**Reply by ondrejholy** [reply]
Thanks.
